### PR TITLE
Ksp 189 user should be able to remove themselves from an event

### PR DESCRIPTION
**Pull request by ivancuadra869**

### CODE REVIEW

File: apps/api/src/modules/etl/events/guests/delete.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+type Context = {
+  firebaseClientInjection: App;
+};
+
+export const etlEventsGuestsDelete = async (
+  params: Params,
+  context: Context,
+  { debug = Debug('api:etl/events/guests/delete') as any } = {}
+) => {
+  assert(params.eventId, makeApiError(409, 'Event is required'));
```
Comment (navarrojandg) [inline apps/api/src/modules/etl/events/guests/delete.ts]: Can you make these 409s into 400 please?

File: apps/api/src/modules/etl/events/guests/delete.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  assert(params.userId, makeApiError(409, 'User is required'));
```
Comment (navarrojandg) [inline apps/api/src/modules/etl/events/guests/delete.ts]: 409 -> 400

File: apps/api/src/modules/etl/events/guests/delete.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      const eventDocRef = firebaseFirestore.doc(`/events/${params.eventId}`);
+      const event = (await transaction.get(eventDocRef)).data();
+
+      assert(event, makeApiError(422, 'Invalid event'));
+
+      const guestsByUserId = event.guestsByUserId as UserId[];
+      // Assert that the params.userId is a guest in the event doc
+      assert(
+        guestsByUserId.includes(params.userId),
+        makeApiError(409, `User: ${params.userId} is not a guest`)
```
Comment (navarrojandg) [inline apps/api/src/modules/etl/events/guests/delete.ts]: 409 -> 400

File: apps/api/src/modules/etl/events/guests/delete.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      const eventGuestsDocRef = firebaseFirestore.doc(
+        `/events/${params.eventId}/guests/${params.userId}`
+      );
+      transaction.delete(eventGuestsDocRef);
+
+      // Updating event guest in events/eventId: { guestsByUserId }
+      const index = guestsByUserId.findIndex(
+        (value) => value === params.userId
+      );
+      guestsByUserId.splice(index, 1);
```
Comment (navarrojandg) [inline apps/api/src/modules/etl/events/guests/delete.ts:62]: A bit unsure if you can splice on a `const` array. So may need to do something like this:

```
transaction.update(eventDocRef, {
  guestsByUserId: event.guestsByUserId.filter(
    (userId) => !userId === params.userId
  ),
});
```

Comment (ivancuadra869) [reply apps/api/src/modules/etl/events/guests/delete.ts:62]: I can change it, but it does currently work

Comment (navarrojandg) [reply apps/api/src/modules/etl/events/guests/delete.ts:62]: Okay if it works then all good

**Review by navarrojandg — APPROVED** [review]
Nice!

File: apps/api/src/modules/etl/events/guests/delete.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      const eventDocRef = firebaseFirestore.doc(`/events/${params.eventId}`);
+      const event = (await transaction.get(eventDocRef)).data();
+
+      assert(event, makeApiError(422, 'Invalid event'));
+
+      const guestsByUserId = event.guestsByUserId as UserId[];
+      // Assert that the params.userId is a guest in the event doc
+      assert(
+        guestsByUserId.includes(params.userId),
+        makeApiError(400, `User: ${params.userId} is not a guest`)
```
Comment (navarrojandg) [inline apps/api/src/modules/etl/events/guests/delete.ts]: Wait my bad this should be 422 -- means the request was properly formatted but there is a semantic / logic error

**Pull request merged**